pkg/nsx/services/subnet/builder.go

```diff
@@ -91,7 +91,14 @@ func (service *SubnetService) buildSubnet(obj client.Object, tags []model.Tag, i
 		if dhcpMode == "" {
 			dhcpMode = v1alpha1.DHCPConfigModeDeactivated
 		}
-		nsxSubnet.SubnetDhcpConfig = service.buildSubnetDHCPConfig(dhcpMode, nil)
+		var dhcpServerAdditionalConfig *model.DhcpServerAdditionalConfig
+		if o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges != nil && len(o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges) > 0 {
+			dhcpServerAdditionalConfig = &model.DhcpServerAdditionalConfig{}
+			for index := range o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges {
+				dhcpServerAdditionalConfig.ReservedIpRanges = append(dhcpServerAdditionalConfig.ReservedIpRanges, o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges[index])
+			}
```

Comment:
Can we just assign the array like `dhcpServerAdditionalConfig.ReservedIpRanges = o.Spec.SubnetDHCPConfig.DHCPServerAdditionalConfig.ReservedIPRanges`?

Reply:
I tried, but with above change, reservedIPRanges is not set.

Reply:
this should work, I changed.

Reply:
I tried after change to "copy“，
--- FAIL: TestBuildSubnetForSubnet (0.00s)
    builder_test.go:213:
        	Error Trace:	/Users/lxiaopei/go_project/nsx-operator/pkg/nsx/services/subnet/builder_test.go:213
        	Error:      	Not equal:
        	            	expected: []string{"10.0.0.4-10.0.0.10"}
        	            	actual  : []string(nil)

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=18) "10.0.0.4-10.0.0.10"
        	            	-}
        	            	+([]string) <nil>

        	Test:       	TestBuildSubnetForSubnet